rest/route/user.go

```diff
@@ -430,15 +446,18 @@ func (h *userRolesPostHandler) Run(ctx context.Context) gimlet.Responder {
 	for _, found := range dbRoles {
 		foundRoles = append(foundRoles, found.ID)
 	}
-	nonexistent, _ := utility.StringSliceSymmetricDifference(h.roles, foundRoles)
+	nonexistent, _ := utility.StringSliceSymmetricDifference(h.Roles, foundRoles)
 	if len(nonexistent) > 0 {
 		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
 			Message:    fmt.Sprintf("roles not found: %v", nonexistent),
 			StatusCode: http.StatusNotFound,
 		})
 	}
-	for _, toAdd := range h.roles {
+	for _, toAdd := range h.Roles {
 		if err = dbUser.AddRole(toAdd); err != nil {
+			grip.Error(message.WrapError(err, message.Fields{
+				"message": "unable to add role",
```

Comment:
Maybe specify which roles/users it couldn't add

Reply:
added